### ISSUE TITLE
Problems with 1-channel tifs

### ISSUE THREAD

**Issue by horsto** [MouseLand/suite2p#82]
This is related to https://github.com/MouseLand/suite2p/issues/81. I encountered problems when running 1 channel tifs through suite2p. Suite2p runs nicely on scanimage tiffs with 2 channels:
![capture](https://user-images.githubusercontent.com/7548928/45253527-306a2200-b369-11e8-9fbb-933dfdb1f605.PNG)

If I extract one channel from that tif and run that through suite2p I get distorted motion correction results (see screenshot below from imagej) and regions are not detected anymore (alignment channel is functional channel in both cases). 
![capture](https://user-images.githubusercontent.com/7548928/45253533-395af380-b369-11e8-956f-b31b7d193666.PNG)

In an unrelated example tif (1 channel, no additional processing) suite2p even tries to tile the image (see screenshot below). 
![capture](https://user-images.githubusercontent.com/7548928/45253549-745d2700-b369-11e8-9d51-e47c63989c3d.PNG)
Any idea where this weirdness comes from?

**Comment by carsen-stringer** [comment]
I was not able to reproduce this issue with my single channel tiffs, are you using the github version of the code? I would recommend using that. When you extract one channel from your tiffs, are you sure that it's the same shape? e.g. try loading it into python
~~~ python
im = skimage.io.imread('*.tif')
print(im.shape)
~~~
suite2p expects the first dim to be time

**Comment by horsto** [comment]
Thanks. Yes, I am using the github code. The shape is correct. 
What seems to have caused problems is that when I exported the 1 channel tif from imagej it was converted from int16 to uint16. If I manually convert the 1 channel tif back to int16  (http://scikit-image.org/docs/dev/api/skimage.html#skimage.img_as_int) and run it through suite2p it seems ok (motion correction ok, same number of ROIs detected). The other, "tiled" example (see last image above) was uint8 to start with. The tiling artefact goes away when I convert it to int16. 

I uploaded the two uint examples here if you want to play with it yourself. 
https://studntnu-my.sharepoint.com/:f:/g/personal/horsto_ntnu_no/Eh9OI7anoy5MmXZkvure6dQBPDvsCSrNdwzfgJEtRSVKNA?e=K4Kbyn

For now I will stick to int16 (all seems fine there!). If the data type is important it might be good to warn people about that.

**Comment by carsen-stringer** [comment]
Ah yeah, the tiff you uploaded has a range from 30000-65000, and int16 only goes up to 32000. I've added the typecasting to the pipeline specifically, so it should in theory work with different data types, but they need to be below 32,000 for things to work out. I will eventually add a rescaling option, I've made an issue for this (issue #87 ).

As a heads up, currently implementing the nonrigid in the pipeline, which should also help with motion correction of that tiff you sent me.
